homeassistant/bootstrap.py

```diff
@@ -69,31 +72,55 @@ def async_setup_component(hass: core.HomeAssistant, domain: str,
         _async_persistent_notification(hass, domain, True)
         return False
 
+    setup_events = hass.data.get(DATA_SETUP)
+    if setup_events is None:
+        hass.data[DATA_PLATFORM] = {}
+        setup_events = hass.data[DATA_SETUP] = {}
+
+    tasks = []
     for component in components:
```

Comment:
I think that we can remove fetching load_order_component completely. `async_setup_component` doesn't have to resolve the whole dependency chain anymore. It will be resolved for us automatically inside `_async_setup_component`.